NitroxServer/Communication/PortForward.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Mono.Nat;
+using NitroxModel.DataStructures;
+
+namespace NitroxServer.Communication
+{
+    public static class PortForward
+    {
+        private static event EventHandler<INatDevice> DeviceDiscovered;
+        private static readonly List<INatDevice> devices = new();
+        private static readonly ThreadSafeDictionary<int, Exception> lastPortErrors = new();
+
+        /// <summary>
+        ///     Tries to open the given port.
+        /// </summary>
+        public static async Task<bool> TryOpenPortAsync(int port, TimeSpan timeout)
+        {
+            static async Task PortForwardDeviceAsync(IDictionary<int, Exception> errors, INatDevice device, int port)
+            {
+                try
+                {
+                    await device.CreatePortMapAsync(new Mapping(Protocol.Udp, port, port, (int)TimeSpan.FromDays(1).TotalSeconds, "Nitrox Server - Subnautica"));
```

Comment:
As it's in the NitroxServer namespace it should probably only be "Nitrox Server".